packages/ember-runtime/lib/computed/reduce_computed_macros.js

```diff
@@ -362,6 +364,49 @@ export function uniq(...args) {
   });
 }
 
+if (isEnabled('ember-runtime-computed-uniq-by')) {
+/**
+  A computed property which returns a new array with all the unique
+  elements from an array, with uniqueness determined by specific key.
+  Example
+  ```javascript
+  var Hamster = Ember.Object.extend({
+    uniqueFruits: Ember.computed.uniqBy('fruits', 'id')
+  });
+  var hamster = Hamster.create({
+    fruits: [
+      { id: 1, 'banana' },
+      { id: 2, 'grape' },
+      { id: 3, 'peach' },
+      { id: 1, 'banana' }
+    ]
+  });
+  hamster.get('uniqueFruits'); // [ { id: 1, 'banana' }, { id: 2, 'grape' }, { id: 3, 'peach' }]
+  ```
+  @method uniqBy
+  @for Ember.computed
+  @param {String} dependentKey
+  @param {String} propertyKey
+  @return {Ember.ComputedProperty} computes a new array with all the
+  unique elements from the dependent array
+  @public
+*/
+export function uniqBy(dependentKey, propertyKey) {
+  var seen = {};
```

Comment:
Stef mentioned this was in error from the last PR. `seen`, defined here, will be shared between two instances of an object.

```js
var MyObject = Ember.Object.extend({
  list: null.
  uniq: Ember.computed.uniqBy('list', 'name')
});

let a = MyObject.create({ list: [{name: 'bob'}, {name: 'mitch'}] });
let b = MyObject.create({ list: [{name: 'warren'}, {name: 'mitch'}] });

a.get('uniq'); // => {name: 'bob', name: 'mitch'}
b.get('uniq'); // => {name: 'warren'}
// ^ This should include mitch, but b/c `seen` is shared it will not.
```

A fix a test for this would be great.